site/docs/plugins/i18n.md

```diff
@@ -391,13 +396,20 @@ interface SessionData {
 
 type MyContext = Context & SessionFlavor<SessionData> & I18nFlavor;
 
+const bot = new Bot<MyContext>(""); // <-- put your bot token here
+
+// Be awared, in Deno Deploy service,
+// we can only load fluent scripts after init I18n.
```

Comment:
Unnecessary imho. The user already has initialized a bot, and the two comments are extra.

```suggestion
```

Reply:
The 2 comment lines are my fault, I supposed to remove it... But if you remove the line 399 the script will be broken if sb try to run it. The `bot` instance is never initialized until this line, isn't it?

Reply:
And that piece of code's goal is not that. It's mentioned how to initialize the bot in the above snippets.

Reply:
I think they will understand if they RTFM from the beginning

Reply:
If you insist.

Reply:
The existing example code snippets fall into two categories:
1) Short snippets. Illustrate a single thing only. One feature, one API call, one way of combining code, etc. These do not include imports and setup code.
2) Long snippets. Illustrate a setup, a complete bot, a minimal copy-paste-run example, etc. These do include imports and setup code.

Reply:
Thanks for clarifying it the formal way, @KnorpelSenf!

Reply:
For me, the same opinion as @ngdangtu-vn. It's not a snippet, it's a full code that include import statements, init session interfaces, init session, a few commands... and we are hidden something that someone will copy this and will not work.

I removed from all languages but 100% disagree, isn't a snippet, is a code to copy-paste and try

Full code:
```
import {
  Bot,
  Context,
  session,
  SessionFlavor,
} from "https://deno.land/x/grammy/mod.ts";
import { I18n, I18nFlavor } from "https://deno.land/x/grammy_i18n/mod.ts";

interface SessionData {
  __language_code?: string;
}

type MyContext = Context & SessionFlavor<SessionData> & I18nFlavor;

const i18n = new I18n<MyContext>({
  defaultLocale: "en",
  useSession: true, // whether to store user language in session

  // DOES NOT work in Deno Deploy
  directory: "locales",
});

// Translation files loaded this way works in Deno Deploy, too.
// await i18n.loadLocalesDir("locales");

// Remember to register `session` middleware before
// registering middleware of the i18n instance.
bot.use(
  session({
    initial: () => {
      return {};
    },
  }),
);

// Register the i18n middleware
bot.use(i18n);

bot.command("start", async (ctx) => {
  await ctx.reply(ctx.t("greeting"));
});

bot.command("language", async (ctx) => {
  if (ctx.match === "") {
    return await ctx.reply(ctx.t("language.specify-a-locale"));
  }

  // `i18n.locales` contains all the locales that have been registered
  if (!i18n.locales.includes(ctx.match)) {
    return await ctx.reply(ctx.t("language.invalid-locale"));
  }

  // `ctx.i18n.getLocale` returns the locale currently using.
  if ((await ctx.i18n.getLocale()) === ctx.match) {
    return await ctx.reply(ctx.t("language.already-set"));
  }

  await ctx.i18n.setLocale(ctx.match);
  await ctx.reply(ctx.t("language.language-set"));
});
```